grails-app/services/org/broadinstitute/orsp/CrowdService.groovy

```diff
@@ -1,131 +1,113 @@
-     *
      * @param filterUsers Collection of userNames to filter on
-     * @param uri Full query url to crowd search
-     * @return List of CrowdUserDetail objects
+     * @return List of CrowdUserDetail objects that do NOT exists on filterUsers
      */
-    private List<CrowdUserDetail> getCrowdQueryUserDetails(Collection<String> filterUsers, String uri) {
+    private List<CrowdUserDetail> getCrowdQueryUserDetails(Collection<String> filterUsers) {
         List<CrowdUserDetail> crowdUsers = new ArrayList<>()
-        def http = new HTTPBuilder(uri)
-        http.request(Method.GET) { req->
-            headers.Accept = 'application/json'
-            headers."User-Agent" = "ORSP" + appInfoConfiguration.version
-            headers.Authorization = "Basic " + getCredentials()
-            response.success = { resp, json ->
-                json.users?.
-                        findAll { it.email && !filterUsers.contains(it.name) }?.
-                        collect(crowdUsers) {
-                            new CrowdUserDetail(
-                                    userName: it.name,
-                                    firstName: it["first-name"],
-                                    lastName: it["last-name"],
-                                    displayName: it["display-name"],
-                                    email: it.email)
-                        }
-            }
 
-            response.failure = { resp ->
-                log.error("Got response: ${resp.statusLine}")
-                log.error("Content-Type: ${resp.headers.'Content-Type'}")
+        // Instantiate a client.
+        BigQuery bigquery =
+                BigQueryOptions.newBuilder()
+                        .setCredentials(getCredential())
+                        .build()
+                        .getService()
+
+        QueryJobConfiguration queryConfig = QueryJobConfiguration
+                .newBuilder("SELECT * FROM `broad-bits.data_warehouse.people`")
+                .setUseLegacySql(false).build()
+
+        // Create a job ID .
+        JobId jobId = JobId.of(UUID.randomUUID().toString());
+        Job queryJob = bigquery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build())
+
+        // Wait for the query to complete.
+        queryJob = queryJob.waitFor()
+
+        // Check for errors
+        if (queryJob == null) {
+            log.error("Job no longer exists")
+        } else if (queryJob.getStatus().getError() != null) {
+            log.error(queryJob.getStatus().getError().toString())
+        } else {
+            // Get the results.
+            TableResult result = queryJob.getQueryResults()
+            // iterate over results to find out missing users
+            for (FieldValueList row : result.iterateAll()) {
+                String email = row.get("broad_email").getStringValue()
+                String userName = row.get("username").getStringValue()
+                if (email && !filterUsers.contains(userName)) {
```

Comment:
To make this class more useful, I would break out these two concepts out into two different functions. By mixing Find-All and Filter in one function, we lose the ability to do more interesting things with the data.

1. Find all Broad Users
2. Filter user list against list of known users

Reply:
makes sense.